### PR TITLE
Anlage in Betrieb nehmen 

### PR DESCRIPTION
**Pull request by lember103**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/model/TimerModel.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+public class TimerModel {
+
+    @JsonProperty("timerDuration")
```
Comment (Haferkorn) [inline backend/src/main/java/de/neuefische/backend/model/TimerModel.java:11]: if attribute and JSONProperty are identical you don't need the annotation

File: backend/src/main/java/de/neuefische/backend/service/TimerService.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Autowired
+    public TimerService(TimerRepo timerRepo) {
+        this.timerRepo = timerRepo;
+    }
+
+    public TimerModel getTimer() {
+        Optional<TimerModel> optionalTimer = timerRepo.findFirstByOrderById();
+
+        if (optionalTimer.isEmpty()){
+            throw new NoSuchElementException("No timer in database");
```
Comment (Haferkorn) [inline backend/src/main/java/de/neuefische/backend/service/TimerService.java:25]: Maybe it is more precise to use a custom Exception

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
-    </>
-  );
+    const [minutes, setMinutes] = useState("")
+
+    const handleClick = () => {
+        postMinutes(minutes).then(response => console.log(response))
+    }
+
+    return (
+        <>
```
Comment (Haferkorn) [inline frontend/src/App.js:13]: for semantics it would be nice to wrap the input field in a form

**Review by Haferkorn — APPROVED** [review]

File: backend/src/main/java/de/neuefische/backend/service/TimerService.java (hunk 2 above)
Comment (lember103) [reply backend/src/main/java/de/neuefische/backend/service/TimerService.java:25]: I changed it to an IllegalStateException which should be more precise in this case.

File: backend/src/main/java/de/neuefische/backend/model/TimerModel.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package de.neuefische.backend.model;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+public class TimerModel {
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/model/TimerModel.java:8]:
```suggestion
public class Timer {
```

File: backend/src/main/java/de/neuefische/backend/model/TimerModel.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+    private String timerDuration;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/model/TimerModel.java:10]: Consider using [Duration class](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html)

File: backend/src/main/java/de/neuefische/backend/service/TimerService.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    public TimerModel getTimer() {
+        Optional<TimerModel> optionalTimer = timerRepo.findFirstByOrderById();
+
+        if (optionalTimer.isEmpty()){
+            throw new IllegalStateException("No timer in database");
+        } else {
+            return optionalTimer.get();
+        }
+    }
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/service/TimerService.java:29]:
```suggestion
    public TimerModel getTimer() {
       return timerRepo.findFirstByOrderById().orElseThrow(()-> new IllegalStateException("No timer in database")
    }
```

File: frontend/src/App.js [hunk 7]
```diff
@@ -1,12 +1,20 @@
...
 
-  return (
-    <>
-      <HelloBackend/>
-    </>
-  );
+    const [minutes, setMinutes] = useState("")
+
+    const handleClick = () => {
+        postMinutes(minutes).then(response => console.log(response))
```
Comment (christophersiem) [inline frontend/src/App.js:9]: why no real post request?

**Review by christophersiem — APPROVED** [review]

File: backend/src/main/java/de/neuefische/backend/model/TimerModel.java (hunk 4 above)
Comment (lember103) [reply backend/src/main/java/de/neuefische/backend/model/TimerModel.java:8]: It's called PumpTimer now

**Pull request merged**